test/widgets/content_test.dart

```diff
@@ -27,6 +27,21 @@ import 'dialog_checks.dart';
 import 'message_list_checks.dart';
 import 'page_checks.dart';
 
+/// Set [debugNetworkImageHttpClientProvider] to return a constant image.
+///
+/// Returns the [FakeImageHttpClient] that handles the requests.
+///
+/// The caller must set [debugNetworkImageHttpClientProvider] back to null
+/// before the end of the test.
+FakeImageHttpClient prepareBoringImageHttpClient() {
```

Comment:
Cool, pulling this out so it can be imported elsewhere makes sense.

It's fine to have this squashed into the main commit as you currently do, because neither this nor the rest of the commit is so long as to start getting harder to read (and because this isn't touching any of the same code, so doesn't interfere with reading the diff for the main changes). But you'd also be welcome to split it out as an NFC prep commit.